### ISSUE TITLE
unlink ENOENT

### ISSUE THREAD

**Issue by errorx666** [log4js-node/log4js-node#959]
log4js 5.3.0:

```
2019-11-04T07:16:24.301 ERROR console internal/process/warning.js:27 (node:122900) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, unlink 'C:\Users\error\Documents\src\error-bot\log\command.2019-10-29.log'
    at emitUnhandledRejectionWarning (internal/process/promises.js:141:15)
    at processPromiseRejections (internal/process/promises.js:203:11)
    at processTicksAndRejections (internal/process/task_queues.js:94:32)

2019-11-04T07:16:24.333 ERROR console internal/process/warning.js:27 (node:122900) Error: ENOENT: no such file or directory, unlink 'C:\Users\error\Documents\src\error-bot\log\command.2019-10-29.log'

2019-11-04T07:16:24.355 ERROR console internal/process/warning.js:27 (node:122900) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
    at emitDeprecationWarning (internal/process/promises.js:153:11)
    at processPromiseRejections (internal/process/promises.js:205:13)
    at processTicksAndRejections (internal/process/task_queues.js:94:32)
```

Config:
```ts
	const logDir = path.resolve( __dirname, '..', 'log' );
	const layout = {
		type: 'pattern',
		pattern: '%d %p %c %f:%l %m%n'
	};

	function logFile( filename: string ) {
		return {
			type: 'dateFile',
			filename: path.resolve( logDir, filename ),
			keepFileExt: true,
			daysToKeep: 30,
			pattern: '.yyyy-MM-dd',
			encoding: 'utf-8',
			layout
		};
	}
	const log4jsConfig = {
		appenders: {
			console: {
				type: 'console',
				layout
			},
			stderr: {
				type: 'stderr',
				layout
			},
			stdout: {
				type: 'stdout',
				layout
			},
			master: logFile( 'master.log' ),
			bus: logFile( 'bus.log' ),
			command: logFile( 'command.log' ),
			query: logFile( 'query.log' ),
			process: logFile( 'process.log' ),
			errors: logFile( 'errors.log' ),
			'only-errors': {
				type: 'logLevelFilter',
				appender: 'errors',
				level: 'error'
			}
		},
		categories: {
			default: {
				appenders: [ 'console', 'master', 'only-errors' ],
				level: 'info',
				enableCallStack: true
			},
			bus: {
				appenders: [ 'bus', 'only-errors' ],
				level: 'info',
				enableCallStack: true
			},
			command: {
				appenders: [ 'command', 'only-errors' ],
				level: 'info',
				enableCallStack: true
			},
			query: {
				appenders: [ 'query', 'only-errors' ],
				level: 'info',
				enableCallStack: true
			},
			process: {
				appenders: [ 'process', 'only-errors' ],
				level: 'debug',
				enableCallStack: true
			}
		},
		disableClustering: true
	};
	log4js.configure( log4jsConfig );
```

**Comment by nomiddlename** [comment]
I published a fix for this in log4js@6.0.0 - please give it a try.

**Comment by errorx666** [comment]
I haven't seen this bug again since upgrading, but it was infrequent anyway.  Closing for now.
